orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandler.kt

```diff
@@ -54,40 +54,23 @@ class CompleteStageHandler(
       if (stage.status in setOf(RUNNING, NOT_STARTED)) {
         var status = stage.determineStatus()
         try {
-          if (status == NOT_STARTED) {
-            // Stage had no before stages or tasks so we'll see if it had any
-            // un-planned after stages
+          if (status == NOT_STARTED || (status.isComplete && !status.isHalt)) {
+            // check to see if this stage has any unplanned synthetic after stages
             var afterStages = stage.firstAfterStages()
             if (afterStages.isEmpty()) {
               stage.planAfterStages()
               afterStages = stage.firstAfterStages()
             }
-            if (afterStages.isNotEmpty()) {
-              afterStages.forEach {
-                queue.push(StartStage(message, it.id))
-              }
-
+            if (afterStages.isNotEmpty() && afterStages.any { it.status == NOT_STARTED }) {
+              afterStages
+                .filter { it.status == NOT_STARTED }
+                .forEach { queue.push(StartStage(message, it.id)) }
               return@withStage
-            } else {
+            } else if (status == NOT_STARTED) {
               // stage had no synthetic stages or tasks, which is odd but whatever
               log.warn("Stage ${stage.id} (${stage.type}) of ${stage.execution.id} had no tasks or synthetic stages!")
               status = SKIPPED
             }
-          } else if (status.isComplete && !status.isHalt) {
-            // check to see if this stage has any unplanned synthetic after stages
-            var afterStages = stage.firstAfterStages()
-            if (afterStages.isEmpty()) {
-              stage.planAfterStages()
-
-              afterStages = stage.firstAfterStages()
-              if (afterStages.isNotEmpty()) {
-                afterStages.forEach {
-                  queue.push(StartStage(message, it.id))
-                }
```

Comment:
The fact that this `queue.push` call was inside the if statement on line 79 was what's been breaking Orca deploys. Old instance pre-planned after stages, new instance therefore didn't run them and just no-opped.